ember/app/utils/download-file.js

```diff
@@ -1,7 +1,10 @@
+import { getOwner } from "@ember/application";
 import { saveAs } from "file-saver";
 
-export default async function downloadFile(uri, init) {
-  const response = await fetch(uri, init);
+export default async function downloadFile(uri, init, context) {
+  const service = getOwner(context).lookup("service:fetch");
+
+  const response = await service.fetch(uri, init);
```

Comment:
If the file response ist already an argument, you don't need to pass the context and fetch arguments in here just to make the requests. 
```js
import { getOwner } from "@ember/application";
import { saveAs } from "file-saver";

export default async function downloadFile(fileResponse) {
  await fileResponse;

  if (!fileResponse.ok) {
    throw new Error(fileResponse.statusText || fileResponse.status);
  }

  const blob = await fileResponse.blob();

  // extract filename from content-disposition header e.g.
  // inline; filename*=utf-8''2022-0948%20-%20Accus%C3%A9%20de%20r%C3%A9ception.pdf
  const filename = decodeURIComponent(
    fileResponse.headers
      .get("content-disposition")
      .match(/filename\*?=['"]?(?:UTF-\d['"]*)?([^;\r\n"']*)['"]?;?/i)[1]
  );

  saveAs(blob, filename);
}
```
Usage:
```js
    try {
      yield downloadFile(this.fetch.fetch(uri, init));
    } catch (error) {
```

If you don't want to do that because of the added fetch lines in you code, i would change this into a service so you can just access the `fetch` service in here.
Something like:
```js
import { getOwner } from "@ember/application";
import { saveAs } from "file-saver";

export default class FileDownloader() {
  @service fetch;

  async function download(uri, init) {
    const response = await this.fetch.fetch(uri, init);

    if (!response.ok) {
      throw new Error(response.statusText || response.status);
    }
    const blob = await response.blob();
    // extract filename from content-disposition header e.g.
    // inline; filename*=utf-8''2022-0948%20-%20Accus%C3%A9%20de%20r%C3%A9ception.pdf
    const filename = decodeURIComponent(
      response.headers
        .get("content-disposition")
        .match(/filename\*?=['"]?(?:UTF-\d['"]*)?([^;\r\n"']*)['"]?;?/i)[1]
    );
    saveAs(blob, filename);
  }
}
```

Reply:
passing promise now